### PR TITLE
Restore "Adds optional deployment key for CI (#10506) (#12590)"

### PR DESCRIPTION
**Pull request by tcsc**
This reverts commit 94e8e4fb4cfa8e5c5107b26a3fae9fda6e871a74.

### CODE REVIEW

**Review by espadolini — APPROVED** [review]
What was the problem with the `teleport.e` CI?

**Comment by tcsc** [comment]
> What was the problem with the `teleport.e` CI?

The build script was still trying to unshallow the `teleport` repo, even though it already had a full copy due to `teleport.e`'s convoluted checkout procedure. See  gravitational/teleport.e#440

File: .cloudbuild/scripts/internal/git/deploykey.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+)
+
+func writeKey(deployKey []byte) (string, error) {
+	// Note that tempfiles are automatically created with 0600, so no-one else
+	// should be able to read this.
+	keyFile, err := os.CreateTemp("", "*")
+	if err != nil {
+		return "", trace.Wrap(err, "failed creating keyfile")
+	}
+	defer keyFile.Close()
```
Comment (jakule) [inline .cloudbuild/scripts/internal/git/deploykey.go:17]: I'd suggest to check for error here. `Write()` goes into memory, `Close()` syncs all data to disk. You may end up with empty file in some rare cases as error returned by `Close()` is ignored.

File: .cloudbuild/scripts/internal/git/knownhosts.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/knownhosts"
+)
+
+func configureKnownHosts(hostname string, keys []ssh.PublicKey) (string, error) {
+	knownHostsFile, err := os.CreateTemp("", "*")
+	if err != nil {
+		return "", trace.Wrap(err, "failed creating known hosts file")
+	}
+	defer knownHostsFile.Close()
```
Comment (jakule) [inline .cloudbuild/scripts/internal/git/knownhosts.go:17]: The same as above. Check for error from `Close()`

**Review by jakule — APPROVED** [review]
Two minor comments. Looks good.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**